src/components/manager/dropBox/DropBoxJsonSelect.tsx

```diff
@@ -0,0 +1,74 @@
+import { useEffect, useState } from "react";
+import { Radio } from "antd";
+
+import { BENTO_DROP_BOX_FS_BASE_PATH } from "@/config";
+import { useDropBoxJsonContent } from "@/modules/dropBox/hooks";
+import { dropBoxTreeNodeEnabledJson } from "@/utils/files";
+
+import DropBoxTreeSelect from "./DropBoxTreeSelect";
+import JsonDisplay from "@/components/display/JsonDisplay";
+import type { JSONType } from "@/types/json";
+
+const InnerDropBoxJsonSelect = ({
+  selectedFile,
+  setSelectedFile,
+}: {
+  selectedFile?: string;
+  setSelectedFile: (x: string) => void;
+}) => {
+  return (
+    <DropBoxTreeSelect
+      basePrefix={BENTO_DROP_BOX_FS_BASE_PATH}
+      multiple={false}
+      nodeEnabled={dropBoxTreeNodeEnabledJson}
+      allowClear={true}
+      value={selectedFile}
+      onChange={(v) => setSelectedFile(v)}
+    />
+  );
+};
+
+export type DropBoxJsonSelectProps = {
+  initialValue?: JSONType;
+  value?: JSONType;
+  onChange?: (x: JSONType) => void;
+};
+
+/** A form input component for DropBox JSON file selection. */
+const DropBoxJsonSelect = ({ initialValue, onChange }: DropBoxJsonSelectProps) => {
+  const editing = initialValue !== undefined;
+
+  const [radioValue, setRadioValue] = useState<"existing" | "new">(editing ? "existing" : "new");
```

Comment:
can add the `none` value as previously discussed - be careful of some ternary logic below which relies on this being a two-value type